File: remove_before_after.py
import os
import re
from pathlib import Path

# Find all service page files
pages_dir = Path("src/pages")
service_files = []

for file_path in pages_dir.rglob("*.astro"):
    # Skip index pages and Home.tsx
    if file_path.name in ["index.astro", "Home.tsx"]:
        continue
    
    # Only process service pages
    if any(service in file_path.name for service in [
        "soft-washing", "roof-cleaning", "pressure-washing", 
        "gutter-cleaning", "upvc-cleaning", "patio-cleaning",
        "driveway-cleaning", "conservatory-cleaning", "brick-cleaning",
        "patio-driveway-cleaning"
    ]):
        service_files.append(file_path)

print(f"Found {len(service_files)} service files to process")

for file_path in service_files:
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original = content
        
        # Remove import statements
        content = re.sub(r"import \{ BeforeAfterSection \} from '@/sections/BeforeAfterSection';\s*\n", "", content)
        content = re.sub(r"import \{ RoofCleaningBeforeAfter \} from '@/sections/BeforeAfterSection/RoofCleaningBeforeAfter';\s*\n", "", content)
        
        # Remove BeforeAfterSection component usage (multiline)
        content = re.sub(r"<!-- .*? Before & After.*?-->\s*<BeforeAfterSection[^>]*>.*?</BeforeAfterSection>\s*", "", content, flags=re.DOTALL)
        content = re.sub(r"<!-- .*? Before & After.*?-->\s*<BeforeAfterSection[^>]*/>\s*", "", content, flags=re.DOTALL)
        
        # Remove RoofCleaningBeforeAfter component usage
        content = re.sub(r"<!-- .*? Before & After.*?-->\s*<RoofCleaningBeforeAfter[^>]*/>\s*", "", content, flags=re.DOTALL)
        
        if content != original:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f"Updated: {file_path.name}")
    except Exception as e:
        print(f"Error processing {file_path}: {e}")

print("Done!")

File: scripts/remove-before-after.js
const fs = require('fs');
const path = require('path');

// Find all service page files
function findServiceFiles(dir, fileList = []) {
  const files = fs.readdirSync(dir);
  
  files.forEach(file => {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);
    
    if (stat.isDirectory()) {
      findServiceFiles(filePath, fileList);
    } else if (file.endsWith('.astro')) {
      // Skip index pages
      if (file === 'index.astro' || file === 'Home.tsx') {
        return;
      }
      
      // Only process service pages
      const serviceNames = [
        'soft-washing', 'roof-cleaning', 'pressure-washing',
        'gutter-cleaning', 'upvc-cleaning', 'patio-cleaning',
        'driveway-cleaning', 'conservatory-cleaning', 'brick-cleaning',
        'patio-driveway-cleaning'
      ];
      
      if (serviceNames.some(name => file.includes(name))) {
        fileList.push(filePath);
      }
    }
  });
  
  return fileList;
}

const pagesDir = path.join(__dirname, '../src/pages');
const serviceFiles = findServiceFiles(pagesDir);

console.log(`Found ${serviceFiles.length} service files to process`);

let updatedCount = 0;

serviceFiles.forEach(filePath => {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    const original = content;
    
    // Remove import statements
    content = content.replace(/import \{ BeforeAfterSection \} from '@/sections\/BeforeAfterSection';\s*\n/g, '');
    content = content.replace(/import \{ RoofCleaningBeforeAfter \} from '@/sections\/BeforeAfterSection\/RoofCleaningBeforeAfter';\s*\n/g, '');
    
    // Remove BeforeAfterSection component usage (multiline)
    content = content.replace(/<!-- [^\r\n]*Before & After[^\r\n]*-->\s*<BeforeAfterSection[^>]*>.*?<\/BeforeAfterSection>\s*/gs, '');
    content = content.replace(/<!-- [^\r\n]*Before & After[^\r\n]*-->\s*<BeforeAfterSection[^>]*\/>\s*/gs, '');
    
    // Remove RoofCleaningBeforeAfter component usage
    content = content.replace(/<!-- [^\r\n]*Before & After[^\r\n]*-->\s*<RoofCleaningBeforeAfter[^>]*\/>\s*/gs, '');
    
    if (content !== original) {
      fs.writeFileSync(filePath, content, 'utf8');
      updatedCount++;
      console.log(`Updated: ${path.basename(filePath)}`);
    }
  } catch (error) {
    console.error(`Error processing ${filePath}:`, error.message);
  }
});

console.log(`\nDone! Updated ${updatedCount} files.`);
